### PR TITLE
Refactor the code

### PR DESCRIPTION
**Pull request by RidmiR**
## Purpose
Refactor the code of MicroRegistryManager. https://github.com/wso2/micro-integrator/pull/544

### CODE REVIEW

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java [hunk 1]
```diff
@@ -123,15 +123,25 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
...
+
+        StringBuilder pathBuilder = new StringBuilder();
+        pathBuilder.append(System.getProperty("carbon.home")).append(File.separator).append("registry").append(File.separator);
+
+        String relativePath = resourcePath.substring(5).replace('/', File.separatorChar);
+        pathBuilder.append("config").append(relativePath);
+
+        boolean isExists = new File(pathBuilder + filename).isFile();
+        try{
+            if (isExists){
```
Comment (chanikag) [inline integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: We can directly return new File(pathBuilder + filename).isFile()

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java [hunk 2]
```diff
@@ -123,15 +123,25 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
...
-        try {
-            String path = sourcePath + resourcePath + filename;
-            return path.contains(filename);
+    public boolean checkResourceExist(String resourcePath, String filename) throws Exception {
+
+        StringBuilder pathBuilder = new StringBuilder();
+        pathBuilder.append(System.getProperty("carbon.home")).append(File.separator).append("registry").append(File.separator);
+
+        String relativePath = resourcePath.substring(5).replace('/', File.separatorChar);
+        pathBuilder.append("config").append(relativePath);
```
Comment (chanikag) [inline integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: have to handle governance paths as well

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java (hunk 1 above)
Comment (chanikag) [reply integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: There are formatting issues and no need to have try-catch block for this condition check

Comment (RidmiR) [reply integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Added the changes. Please review.

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java (hunk 2 above)
Comment (RidmiR) [reply integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Added the changes. Please review.

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java [hunk 3]
```diff
@@ -123,19 +123,33 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
...
+        } else if (resourcePath.startsWith(GOV_PATH)) {
+            String relativePath = resourcePath.substring(4).replace('/', File.separatorChar);
+            pathBuilder.append("governance").append(relativePath);
+        } else {
+            throw new MicroRegistryManagerException("Exception occurred while checking the registry resource file: " + resourcePath);
+        }
 
-        } catch (Exception e) {
-            throw new Exception("Exception occurred while checking the registry resource file.", e);
+        if(new File(pathBuilder + filename).isFile()){
```
Comment (chanikag) [inline integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Shall we directly return new File(pathBuilder + filename).isFile() 
Then we can omit this condition check

**Comment by RidmiR** [comment]
Closing this PR as mistakenly added few git commands. Created a new PR. https://github.com/wso2/micro-integrator/pull/589

**Pull request closed without merging**